halo2_proofs/src/dev.rs

```diff
@@ -476,7 +491,7 @@ impl<F: Field> Assignment<F> for MockProver<F> {
     }
 }
 
-impl<F: Field + Ord> MockProver<F> {
+impl<F: PrimeField + Ord> MockProver<F> {
```

Comment:
Logically this shouldn't need to be a prime field because, for example, we might want to support proving systems that are not group-based, such as FRI. (For a group-based proving system, a non-prime field would be insecure because discrete log on the group could be broken by Pohlig–Hellman.)